Description: Prefer a /usr/share directory named after the package if it
  exists.
Author: Chad Miller <chad.miller@canonical.com>

Index: beta/chrome/common/chrome_paths.cc
===================================================================
--- beta.orig/chrome/common/chrome_paths.cc
+++ beta/chrome/common/chrome_paths.cc
@@ -66,6 +66,9 @@ const base::FilePath::CharType kFilepath
 // where it can find the latest component updated flash.
 const base::FilePath::CharType kComponentUpdatedFlashHint[] =
     FILE_PATH_LITERAL("latest-component-updated-flash");
+
+const base::FilePath::CharType kFilepathSinglePrefExtensionsNew[] =
+    FILE_PATH_LITERAL("/usr/share/chromium-browser/extensions");
 #endif  // defined(OS_LINUX) || defined(OS_CHROMEOS)
 
 #if (defined(OS_LINUX) || defined(OS_CHROMEOS)) && \
@@ -496,6 +499,9 @@ bool PathProvider(int key, base::FilePat
       cur = base::FilePath(FILE_PATH_LITERAL("/etc/opt/chrome/policies"));
 #else
       cur = base::FilePath(FILE_PATH_LITERAL("/etc/chromium/policies"));
+      if (!base::PathExists(cur)) {
+        cur = base::FilePath(FILE_PATH_LITERAL("/etc/chromium-browser/policies"));
+      }
 #endif
       break;
     }
@@ -511,6 +517,10 @@ bool PathProvider(int key, base::FilePat
 #endif
 #if defined(OS_LINUX) || defined(OS_CHROMEOS)
     case chrome::DIR_STANDALONE_EXTERNAL_EXTENSIONS: {
+      if (access(kFilepathSinglePrefExtensionsNew, R_OK|X_OK) == 0) {
+        cur = base::FilePath(kFilepathSinglePrefExtensionsNew);
+        break;
+      }
       cur = base::FilePath(kFilepathSinglePrefExtensions);
       break;
     }
@@ -561,6 +571,10 @@ bool PathProvider(int key, base::FilePat
 #else
       cur = base::FilePath(FILE_PATH_LITERAL(
           "/etc/chromium/native-messaging-hosts"));
+      if (!base::PathExists(cur))
+        cur = base::FilePath(FILE_PATH_LITERAL(
+            "/etc/chromium-browser/native-messaging-hosts"));
+
 #endif
 #endif  // !defined(OS_MAC)
       break;
